### PR TITLE
V3.0.0

### PR DESCRIPTION
**Pull request by gnelissen**
New major release of the SAG analysis framework. 
- It integrates the analysis of DAG, self-suspending and EDD tasks as presented in the paper 
S. Srinivasan, M. Gunzel, and G. Nelissen, “Response-Time Analysis of for Limited-Preemptive Self-Suspending and Event-Driven Delay-Induced Tasks”, Proceedings of the 45th IEEE Real-Time Systems Symposium (RTSS 2024), to appear.

- It implements major code changes (credit to Srinidhi Srinivasan, Richard Verhoeven, and Geoffrey Nelissen) that accelerates the analysis by several orders of magnitude. The actual gain depends on the specific system being analyzed. Larger more complicated problems tend to show higher speedups.

- It modifies the data structure used in `index_set` to address a bottleneck in the code that only appeared on Linux (credit Pourya Gohari). The old data structure could result in slowing the analysis down up to 40% in the version 2.2.0 and by more than a 100X in the new version 3.0.0 even on some simple problems. That slow down was only experienced on Linux. Windows and macOS were not affected.

### CODE REVIEW

File: scripts/analyze-global-dag-task-set.sh
[the remark this answers is not in the document]
Comment (gnelissen) [reply scripts/analyze-global-dag-task-set.sh:1]: it seems white spaces have been changed in them by someone...

**Comment by gnelissen** [comment]
> This test has an issue when the code is compiled with the parallel option. As can be seen in the actions, the `./runtests` didn't finish because it took a very long time. I see the same behavior on my laptop.

I cannot reproduce that issue. I work on Windows and all the tests are passing on my machine even with PARALLEL_RUN=ON

**Comment by porya-gohary** [comment]
> > This test has an issue when the code is compiled with the parallel option. As can be seen in the actions, the `./runtests` didn't finish because it took a very long time. I see the same behavior on my laptop.
> 
> I cannot reproduce that issue. I work on Windows and all the tests are passing on my machine even with PARALLEL_RUN=ON

I can replicate the issue on my laptop and try to solve it. But I just noticed another problem. This test is not checking anything; it is just comparing the results of the same exact function. I don't think this is a proper test... If I want to guess probably this test is written before removing the uni space.

**Comment by porya-gohary** [comment]
@gnelissen I have implemented a patch to address the parallel execution issue. Could you please review and verify the changes?

**Comment by gnelissen** [comment]
the changes are correct and indeed catch a potential bug in the parallel version of the code.

**Pull request merged**